src/library/coverartcache.cpp

```diff
@@ -7,9 +7,42 @@
 #include "library/coverartcache.h"
 #include "library/coverartutils.h"
 
-// Large cover art wastes space in our cache when we typicaly won't show them at
-// their full size. If no width is specified, this is the maximum width cap.
-const int kMaxCoverWidth = 300;
+
+namespace {
+    QString pixmapCacheKey(quint16 hash, int width) {
+        if (width == 0) {
+            return QString("CoverArtCache_%1").arg(QString::number(hash));
+        } else {
+            return QString("CoverArtCache_%1_%2")
+                    .arg(QString::number(hash)).arg(width);
+        }
+    }
+
+    // Large cover art wastes space in our cache when we typically won't show them at
+    // their full size. If no width is specified, this is the maximum width cap.
+    const int kMaxCoverSize = 300;
+
+    // The transformation mode when scaling images
+    const Qt::TransformationMode kTransformationMode = Qt::SmoothTransformation;
+
+    // Resizes the image (preserving aspect ratio) if it is larger than
+    // maxEdgeSize on either side.
+    inline QImage limitImageSize(const QImage& image, int maxSize) {
```

Comment:
Since it is "only" a 280 line PR this will be not be required. 
I will read it line by line. For similar PRs it would be nice to have the moves separate.